src/WidgetBase.ts

```diff
@@ -71,6 +69,23 @@ export function diffProperty(propertyName: string, diffType = DiffType.CUSTOM, d
 }
 
 /**
+ * Decorator that can be used to enable property diffing of all properties
+ */
+export function diffProperties(constructor: Function): void;
```

Comment:
Either way we should probably just decide a style, I know this is overloaded but with no params I'd imagine users would expect it to be like `@afterRender` and `onPropertiesChanged@`, which don't take parenthesis.